server/datastore/mysql/policies.go

```diff
@@ -690,3 +690,128 @@ func (ds *Datastore) CleanupPolicyMembership(ctx context.Context, now time.Time)
 
 	return nil
 }
+
+func (ds *Datastore) IncrementPolicyViolationDays(ctx context.Context, now time.Time) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		return incrementViolationDaysDB(ctx, tx, now)
+	})
+}
+
+func incrementViolationDaysDB(ctx context.Context, tx sqlx.ExtContext, now time.Time) error {
+	const (
+		statsID        = 0
+		statsType      = "policy_violation_days"
+		updateInterval = 24 * time.Hour
+	)
+
+	// get current count of policy violation days from `aggregated_stats``
+	selectStmt := `
+		SELECT 
+			json_value, 
+			created_at, 
+			updated_at 
+		FROM 
+			aggregated_stats 
+		WHERE 
+			id = ? AND type = ?`
+	selectResult := []struct {
+		CreatedAt time.Time `json:"created_at" db:"created_at"`
+		UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
+		Count     uint      `json:"json_value" db:"json_value"`
+	}{}
+	if err := sqlx.SelectContext(ctx, tx, &selectResult, selectStmt, statsID, statsType); err != nil {
+		return ctxerr.Wrap(ctx, err, "selecting policy violation days aggregated stats")
+	}
+
+	var prevCount uint
+	var shouldIncrement bool
+	switch len(selectResult) {
+	case 0:
+		// no previous count exists so initialize count as zero and proceed to increment
+		prevCount = 0
+		shouldIncrement = true
+	case 1:
+		// increment previous count if interval has elapsed
+		prevCount = selectResult[0].Count
+		lastUpdated := selectResult[0].UpdatedAt
+		if selectResult[0].CreatedAt.After(selectResult[0].UpdatedAt) {
```

Comment:
I would've expected updated_at to be the same as created_at by default, so this condition should never be true. If that's not the case, I would try to solve this at the sql level instead of here.

Reply:
That was my thought initially as well but then I found this [pattern](https://github.com/fleetdm/fleet/blob/5234039bb6d9a3b42a1ddddb5c35398a74d2edb0/server/datastore/mysql/statistics.go#L117) in the `statistics` code. Do you recall why it was needed there?

Reply:
I do not recall, no.